app/services/sso/openid_connect.rb

```diff
@@ -4,27 +4,48 @@ class OpenidConnect < Base
     attr_reader :current_user
 
     def available?
+      if session[:sso_method] == "SSO::OpenidConnect"
+        session.delete(:sso_method) unless session[:user]
+        return true
+      end
       ((controller.api_request? && bearer_token_set?) || http_token.present?) && valid_issuer?
     end
 
     def authenticate!
+      self.has_rendered = true
+      controller.redirect_to login_url
+    end
+
+    def authenticated?
       payload = jwt_token.decode
-      return nil if payload.nil?
+      return false if payload.nil?
       user = find_or_create_user_from_jwt(payload)
       @current_user = user
       update_session(payload)
       user&.login
-    end
-
-    def authenticated?
-      self.user = User.current.presence || authenticate!
+      self.user = User.current.presence
+      true
```

Comment:
Should this be `User.current.present?` instead of `true` to make sure the user has actually been properly set?

Reply:
Frankly, I dont understand how the User.current is being set. And while testing it, User.current is always false.

Reply:
Then I guess best to leave out from here and just call it after `post_successful_login`.